discord/client.pyi

```diff
@@ -210,8 +210,8 @@ class Client:
                  timeout: Optional[float] = ...) -> asyncio.Future[Tuple[Role, Role]]: ...
     @overload
     def wait_for(self, event: Literal['guild_emojis_update'], *,
-                 check: Optional[Callable[[Guild, List[Emoji], List[Emoji]], bool]] = ...,
-                 timeout: Optional[float] = ...) -> asyncio.Future[Tuple[Guild, List[Emoji], List[Emoji]]]: ...
+                 check: Optional[Callable[[Guild, Tuple[Emoji], Tuple[Emoji]], bool]] = ...,
```

Comment:
If this is a variable-length Tuple it should be `Tuple[Emoji, ...]`. But how come the type hints use `Tuple` when the documentation says `Sequence`?

Reply:
Good catch. I'm not sure why the docs are `Sequence` because what is produced are `Tuple[Emoji, ...]` (the event emits the before and after state of `Guild.emojis`) so I made the typings reflect the code.

Reply:
It was changed to sequence in a [recent PR](https://github.com/Rapptz/discord.py/pull/2370) which corrected the docstring, as it was wrong before.

Reply:
`Sequence` is generally used (in typings) for an argument where you don't care about the exact type, but only that it implements that protocol. It's not generally used in return statements/callback arguments where the type is known. If the sequence of emojis could at some point be a list and another point a tuple, I could see using `Sequence`, however the sequnce of emojis will [only ever be a tuple](https://github.com/Rapptz/discord.py/blob/master/discord/state.py#L665).